src/lib/fastDownward/missingSkillDetection.ts

```diff
@@ -127,25 +128,25 @@ export function skillAnalysis<LU extends LearningUnit>(
         }
     }
     
-    // If goalString still has some goals, then we check the skill groups in globalKnowledge
+    // If goalsArray still has some goals, then we check the skill groups in globalKnowledge
     // To remove parent skill if all his children is found (Reachable)
-    // We do loop over the goal skills, to void the order of the skills until no more changes on the goalString
-    if (goalString.length > 0) {
+    // We do loop over the goal skills, to void the order of the skills until no more changes on the goalsArray
+    if (goalsArray.length > 0) {
         while (true) {
-            const goalSkills = goalString.split(",").filter(skill => skill.length > 0);
+            const goalSkills = goalsArray.slice(0);
             let goalSkillsCount = goalSkills.length;
             goalSkills.forEach(sk => {
                 const parentSkill = globalKnowledge.getAllParents().find(parentSkill => parentSkill.id == sk);
                 if (parentSkill) {
                     let parentSkillCount = parentSkill.nestedSkills.length;
                     parentSkill.nestedSkills.forEach(skill => {
-                            if (goalString.includes(`,`.concat(skill).concat(`,`))) {
+                            if (goalsArray.includes(skill)) {
                                 parentSkillCount--;
                             }
                         }
                     );
                     if (parentSkillCount == parentSkill.nestedSkills.length) {
-                        goalString = goalString.replace(`,${sk},`,`,`);
+                        goalsArray.splice(goalsArray.indexOf(sk), 1)
                         goalSkillsCount--;
                     }

```

Comment:
hard to read code with a lot of complexity and nested loops and conditions. especially the `while(true)` could be hard to understand. 

I'm aware that this was introduced via this pull requests, but this should be addressed in the future.